cmake/Platform/Mac/InstallUtils_mac.cmake.in

```diff
@@ -0,0 +1,149 @@
+        ${framework_path}/Python
+        ${framework_path}/Resources
+        ${framework_path}/Headers
+    )
+    
+    file(GLOB_RECURSE exe_file_list "${framework_path}/**/*.exe")
+    if(exe_file_list)
+        file(REMOVE_RECURSE ${exe_file_list})
+    endif()
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink include/python3.7m Headers
+        WORKING_DIRECTORY ${framework_path}/Versions/3.7
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink 3.7 Current
+        WORKING_DIRECTORY ${framework_path}/Versions/
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Python Python
+        WORKING_DIRECTORY ${framework_path}
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Headers Headers
+        WORKING_DIRECTORY ${framework_path}
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Resources Resources
+        WORKING_DIRECTORY ${framework_path}
+    )
+    file(CHMOD ${framework_path}/Versions/Current/Python 
+        PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE 
+    )
+
+endfunction()
+
+function(codesign_file file entitlement_file)
+
+    if(EXISTS ${entitlement_file})
+
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "--entitlements" "${entitlement_file}" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    else()
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    endif()
+
+    if(NOT ${codesign_ret} EQUAL "0")
+        message("Codesign operation for ${file_path} returned ${codesign_ret} with message ${codesign_out}")
+    endif()
+
+endfunction()
+
+function(codesign_python_framework_binaries framework_path)
+
+    # The codesign "--deep" flag will only codesign binaries in folders with specific names.
+    # We need to codesign all the binaries that the "--deep" flag will miss.
+    file(GLOB_RECURSE files
+        LIST_DIRECTORIES false
+        "${framework_path}/Versions/3.7/bin/**"
+        "${framework_path}/Versions/3.7/lib/**"
+        "${framework_path}/Versions/3.7/Resources/**")
+
+    foreach(file ${files})
+        if(NOT EXISTS ${file})
+            file(REMOVE ${file})
+            continue()
+        endif()
+        cmake_path(SET path_var "${file}")
+        cmake_path(GET path_var EXTENSION LAST_ONLY extension)
+        set(should_codesign FALSE)
+        if (NOT extension)
+            set(should_codesign TRUE)
+        elseif(${extension} STREQUAL ".dylib" OR ${extension} STREQUAL ".so" OR ${extension} STREQUAL ".7m")
```

Comment:
Using IN_LIST, forces me to enable CMP0057. Both this and the other policy issue are happening because there's no min version of CMake specified anywhere in the install script(like we do in our root CMakeLists.txt). Adding `install(CODE "cmake_minimum_required(VERSION 3.20)")` in ly_setup_o3de_install() fixes the issue. I thought of adding it there because it may be useful for other platforms in the future and customers would need that version to build anyway.